validator/validator_test.go

```diff
@@ -0,0 +1,231 @@
+package validator
+
+import (
+	"testing"
+
+	"github.com/giantswarm/inago/controller"
+)
+
+// TestValidateRequest tests the ValidateRequest function.
+func TestValidateRequest(t *testing.T) {
+	var tests = []struct {
+		request controller.Request
+		valid   bool
+		err     error
+	}{
+		{
+			request: controller.Request{
+				Group: "empty",
+			},
+			valid: false,
+			err:   noUnitsInGroupError,
+		},
+		{
+			request: controller.Request{
+				Group: "single",
+				Units: []controller.Unit{
+					controller.Unit{
+						Name: "single-unit.service",
+					},
+				},
+			},
+			valid: true,
+			err:   nil,
+		},
+		{
+			request: controller.Request{
+				Group: "single",
+				Units: []controller.Unit{
+					controller.Unit{
+						Name: "single-unit.service",
+					},
+					controller.Unit{
+						Name: "single-unit2.timer",
+					},
+				},
+			},
+			valid: true,
+			err:   nil,
+		},
+		{
+			request: controller.Request{
+				Group: "scalable",
+				Units: []controller.Unit{
+					controller.Unit{
+						Name: "scalable-unit@.service",
+					},
+				},
+			},
+			valid: true,
+			err:   nil,
+		},
+		{
+			request: controller.Request{
+				Group: "scalable",
+				Units: []controller.Unit{
+					controller.Unit{
+						Name: "scalable-unit@.service",
+					},
+					controller.Unit{
+						Name: "scalable-unit2@.timer",
+					},
+				},
+			},
+			valid: true,
+			err:   nil,
+		},
+		{
+			request: controller.Request{
+				Group: "mix",
+				Units: []controller.Unit{
+					controller.Unit{
+						Name: "mix-unit1.service",
+					},
+					controller.Unit{
+						Name: "mix-unit2@.service",
+					},
+				},
+			},
+			valid: false,
+			err:   mixedSliceInstanceError,
+		},
+		{
+			request: controller.Request{
+				Group: "single",
+				Units: []controller.Unit{
+					controller.Unit{
+						Name: "bad-prefix.service",
+					},
+				},
+			},
+			valid: false,
+			err:   badUnitPrefixError,
+		},
+		{
+			request: controller.Request{
+				Group: "bad@groupname@",
+				Units: []controller.Unit{
+					controller.Unit{
+						Name: "bad@groupname@.service",
+					},
+				},
+			},
+			valid: false,
+			err:   atInGroupNameError,
+		},
+		{
+			request: controller.Request{
+				Group: "group",
+				Units: []controller.Unit{
+					controller.Unit{
+						Name: "group-un@it@.service",
+					},
+				},
+			},
+			valid: false,
+			err:   multipleAtInUnitNameError,
+		},
+		{
+			request: controller.Request{
+				Group: "group",
+				Units: []controller.Unit{
+					controller.Unit{
+						Name: "group-unit1@.service",
```

Comment:
This lines look pretty much the same. A comment would be nice to break the silence. Somebody may overlooks the difference of the unit names.